website/content/en/preview/concepts/threat-model.md

```diff
@@ -0,0 +1,87 @@
+| Generic	| Karpenter has permissions to create, modify, and delete nodes from the cluster, and evict any pod. 	| Cluster Operators running applications with varying security profiles in the same cluster may want to configure dedicated nodes and scheduling rules for Karpenter to mitigate potential container escapes from other containers	|
+| AWS-Specific	| The Karpenter IAM policy is encoded in the GitHub repo. Any additional permissions possibly granted to that role by the administrator are out of scope	|	|
+| AWS-Specific	| The Karpenter pod uses IRSA for AWS credentials 	| Setup of IRSA is out of scope for this document 	|
+
+## Generic Threats and Mitigations
+
+### Threat: Cluster Developer can influence creation of an arbitrary number of nodes
+
+**Background**: Karpenter creates new instances based on the count of pending pods.
+
+**Threat**: A Cluster Developer attempts to have Karpenter create more instances than intended by creating a large number of pods or by using anti-affinity to schedule one pod per node.
+
+**Mitigation**: In addition to [Kubernetes resource limits](https://kubernetes.io/docs/concepts/policy/resource-quotas/#object-count-quota), Cluster Operators can [configure limits on a Provisioner](https://karpenter.sh/preview/concepts/provisioners/#speclimitsresources) to limit the total amount of memory, CPU, or other resources provisioned across all nodes.
+
+## AWS-Specific Threats
+
+### Threat: Launching EC2 instances with IAM roles not intended for Karpenter nodes
+
+**Background**: Many IAM roles in an AWS account may trust the EC2 service principal. IAM administrators must grant the `iam:PassRole` permission to IAM principals to allow those principals in the account to launch instances with specific roles.
+
+**Threat:** A Cluster Operator attempts to create a Node Template with an IAM role not intended for Karpenter
+
+**Mitigation**: Cluster Operators must enumerate the roles in the resource section of the IAM policy granted to the Karpenter role for the `iam:PassRole` action.
+
+### Threat: Karpenter can be used to create or terminate EC2 instances outside of the cluster
+
+**Background**: EC2 instances can exist in an AWS account outside of the Kubernetes cluster.
+
+**Threat**: An actor who obtains control of the Karpenter pod’s IAM role may be able to create or terminate EC2 instances not part of the Kubernetes cluster managed by Karpenter.
+
+**Mitigation**: Karpenter creates instances with tags, several of which can be enforced in the IAM policy granted to the Karpenter IAM role that restrict the instances Karpenter can terminate. One tag can require that the instance was provisioned by a Karpenter controller, another tag can include a cluster name to mitigate any termination between of two clusters with Karpenter in the same account. Cluster Operators also can restrict the region to prevent two clusters in the same account with the same name in different regions.
```

Comment:
```suggestion
**Mitigation**: Karpenter creates instances with tags, several of which can be enforced in the IAM policy granted to the Karpenter IAM role that restrict the instances Karpenter can terminate. One tag can require that the instance was provisioned by a Karpenter controller, another tag can include a cluster name to mitigate any termination between two clusters with Karpenter in the same account. Cluster Operators also can restrict the region to prevent two clusters in the same account with the same name in different regions.
```